ZuluContent/Zulu/Engines/Magic/EnchantmentExtensions.cs

```diff
@@ -116,6 +117,19 @@ public static bool TrySetResist(this IEnchanted enchanted, ElementalType type, i
         public static bool TrySetResist<T>(this T mobile, ElementalType type, ElementalProtectionLevel level)
             where T : Mobile, IEnchanted => TrySetResist((IEnchanted) mobile, type, level);
 
+        public static PoisonLevel GetPoisonResist(IEnchanted ench)
```

Comment:
Couldn't figure out how to do this inline for both PoisonImmunity (the buff) and PoisonProtection. Is there a way we could combine the 2?

Reply:
Might be possible by chaining the null-coalescing operator

```csharp
return ench.GetDistinctEnchantment<PoisonImmunity>()?.Value 
    ?? ench.GetDistinctEnchantment<PoisonProtection>()?.Value 
    ?? 0 as PoisonLevel;
```

Reply:
Yup this worked without the cast being needed, thanks :)